src/main/java/de/tum/cit/aet/artemis/programming/web/localci/BuildLogResource.java

```diff
@@ -52,4 +55,25 @@ public ResponseEntity<Resource> getBuildLogForBuildJob(@PathVariable String buil
         responseHeaders.setContentDispositionFormData("attachment", "build-" + buildJobId + ".log");
         return new ResponseEntity<>(buildLog, responseHeaders, HttpStatus.OK);
     }
+
+    /**
+     * GET /build-log/{buildJobId}/entries : get the build log entries for a given result
+     *
+     * @param buildJobId the id of the build job for which to retrieve the build log entries
+     * @return the ResponseEntity with status 200 (OK) and the build log entries in the body, or with status 404 (Not Found) if the build log entries could not be found
+     */
+    @GetMapping("build-log/{buildJobId}/entries")
+    @EnforceAtLeastEditor
+    public ResponseEntity<List<BuildLogDTO>> getBuildLogEntriesForBuildJob(@PathVariable String buildJobId) {
```

Comment:
Can this endpoint be used by every editor (across courses) or only by editors of courses to which this build job belongs?

Reply:
I am using the same authorization used in `getBuildLogForBuildJob` (method to download the logs). I would have to change the endpoint to include the course id to be able to use annotation, Otherwise I would have to load the buildjob from database just to check for the course id. I don't think this is necessary as it's pretty difficult to figure out the buildjobId (it's based on the timestamp of time the buildjob was queued)